manila/share/drivers/netapp/dataontap/cluster_mode/lib_multi_svm.py

```diff
@@ -611,17 +622,24 @@ def _delete_vserver_without_lock():
             self._client.delete_vserver(vserver,
                                         vserver_client,
                                         security_services=security_services)
-            ipspace_deleted = False
-            if (ipspace_name and ipspace_name not in CLUSTER_IPSPACES
-                    and not self._client.ipspace_has_data_vservers(
-                        ipspace_name)):
-                self._client.delete_ipspace(ipspace_name)
-                ipspace_deleted = True
-
-            if not ipspace_name or ipspace_deleted:
-                # NOTE(dviroel): only delete vlans if they are not being used
-                # by any ipspaces and data vservers.
-                self._delete_vserver_vlans(interfaces_on_vlans)
+
+            ipspace_deleted = self._client.delete_ipspace(ipspace_name)
+
+            # NOTE(dviroel): only delete vlans if they are not being used
+            # by any ipspaces and data vservers.
+            if ipspace_name and not ipspace_deleted:
+                # NOTE(carthaca): filter for degraded ports only, those where
+                # reachability is down (e.g. because neutron port is gone),
+                # intentionally overwriting all ports
+                self._vserver_ports[vserver] = self._client.get_degraded_ports(
+                    broadcast_domains, ipspace_name)
+                # make sure to delete ports of the vserver we are currently
+                # deleting (may not be marked degraded, yet)
+                for interface in interfaces_on_vlans:
+                    port = f"{interface['home-node']}:{interface['home-port']}"
+                    self._vserver_ports[vserver].append(port)
+
+            self._delete_port_vlans(self._client, self._vserver_ports[vserver])
 
         @utils.synchronized('netapp-VLAN-%s' % vlan_id, external=True)
         def _delete_vserver_with_lock():
```

Comment:
Just a comment. what's the purpose of locking? In case, there are multiple vlans in the ipspaces, the lock uses the vlan id belonging to the to-be-deleted vserver. It's probably ok, because any parallel operation on the degraded vlan is not very likely anyway.

Reply:
I agree. I think setup_server is supposed to wait for delete_server and it is okay to do this only for the non-orphaned one